### PR TITLE
Initial implementation of prefetched buffer, convert DRAM and pattern_arith to use it

### PR DESCRIPTION
**Pull request by illuhad**
Add `PrefetchedBuffer` which is a buffer wrapper to make it easier to make measurements without including data transfers.

I'm still working on converting the entire suite to use it (so far I've only converted DRAM and pattern_arith to use it), so WIP.

### CODE REVIEW

File: include/prefetched_buffer.h [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  {
+    return buff->template get_access<mode>(accessRange, accessOffset);
+  }
+
+  cl::sycl::buffer<T, Dimensions>& get() const
+  { return *buff; }
+
+private:
+  // Wrap in a shared_ptr to allow default constructing this class
+  std::shared_ptr<cl::sycl::buffer<T, Dimensions>> buff;
```
Comment (psalz) [inline include/prefetched_buffer.h]: Why not unique?

File: micro/DRAM.cpp [hunk 2]
```diff
@@ -32,20 +32,16 @@ class MicroBenchDRAM {
   // Since we cannot use explicit memory operations to initialize the input buffer,
   // we have to keep this around, unfortunately.
   std::vector<DataT> input;
```
Comment (psalz) [inline micro/DRAM.cpp]: If passing a `const DataT*` pointer forces a copy (as we discussed yesterday), you could also get rid of this here and only temporarily create it in `setup()`.

File: include/prefetched_buffer.h [hunk 3]
```diff
@@ -0,0 +1,74 @@
+#pragma once
```
Comment (psalz) [inline include/prefetched_buffer.h]: Maybe run this file through `clang-format`..? :)

File: include/prefetched_buffer.h (hunk 1 above)
Comment (illuhad) [reply include/prefetched_buffer.h]: That would make `PrefetchedBufffer` non-copyable. I wanted it to have the same semantics as regular buffer in order to have code break at as few places as possible when we do the transition.

File: include/prefetched_buffer.h (hunk 3 above)
Comment (illuhad) [reply include/prefetched_buffer.h]: Will do!

File: include/prefetched_buffer.h (hunk 1 above)
Comment (psalz) [reply include/prefetched_buffer.h]: I see, makes sense!

**Pull request merged**